src/client/app/strings/english.ts

```diff
@@ -40,6 +41,7 @@ export function getEnglishStrings(): ClientStrings {
         snippetImportErrorBody: `We couldn't import the snippet.`,
         reloadPrompt: 'Reload this task pane and then try another URL or ID.',
 
+        snippetNameDuplicationError: 'There is an existing snippet with this name. Do you want to go to this snippet or create a new one with a different name?',
```

Comment:
Also, now that I think of it, I think we should do it for all GISTS, regardless of whether the user owns them or if he/she is importing someone else's.  Not sure if your current code covers that or not.

Could you add that to the manual test scenarios too?  (I.e., try to import the same external snippet twice.  Vs try to import your own snippet twice).

But note, only gists, not samples or anything else.  It should be totally reasonable for someone to import multiple copies of the same sample (esp. a "basic api sample").